packages/odyssey-storybook/src/components/odyssey-labs/AppSwitcher/AppSwitcher.stories.tsx

```diff
@@ -0,0 +1,145 @@
+import { PlaywrightProps } from "../../odyssey-mui/storybookTypes";
+
+const storybookMeta: Meta<AppSwitcherProps> = {
+  title: "Labs Components/AppSwitcher",
+  component: AppSwitcher,
+  argTypes: {
+    appIcons: {
+      description: "",
+      table: {
+        type: {
+          summary: "Array<AppSwitcherAppIcon>",
+        },
+      },
+    },
+    isLoading: {
+      control: "boolean",
+    },
+    selectedAppName: {
+      description: "The App Instance app name that the UI is on",
+      control: "text",
+    },
+  },
+  args: {
+    appIcons: [],
+  },
+  decorators: [MuiThemeDecorator],
+  parameters: {
+    layout: "fullscreen",
+  },
+};
+
+export default storybookMeta;
+
+type Story = StoryObj<AppSwitcherProps>;
+
+export const AllApps: Story = {
+  args: {
+    appIcons: [
+      {
+        appIconDefaultUrl: "/appswitcher/admin-app-default.svg",
+        appIconSelectedUrl: "/appswitcher/admin-app-selected.svg",
+        appName: "saasure",
+        label: "Admin Dashboard",
+        linkUrl: "http://rain-admin.okta1.com:1802/admin/dashboard",
+      },
+      {
+        appIconDefaultUrl: "/appswitcher/okta-dashboard-default.svg",
+        appIconSelectedUrl: "/appswitcher/okta-dashboard-selected.svg",
+        appName: "okta_enduser",
+        label: "Okta Dashboard",
+        linkUrl: "http://rain.okta1.com:1802/app/UserHome",
+      },
+      {
+        appIconDefaultUrl: "/appswitcher/workflows-default.svg",
+        appIconSelectedUrl: "/appswitcher/workflows-selected.svg",
+        appName: "okta_flow_sso",
+        label: "Okta Workflows",
+        linkUrl: "http://rain-admin.okta1.com:1802/flow/go",
+      },
+    ],
+    isLoading: false,
+    selectedAppName: "saasure",
+  },
+  render: (props) => {
+    return (
+      <div style={{ height: "100vh" }}>
+        <AppSwitcher {...props} />
+      </div>
+    );
+  },
+  play: async ({ canvasElement, step }: PlaywrightProps<AppSwitcherProps>) => {
+    const canvas = within(canvasElement);
+
+    await step("App switcher shows all items", async () => {
+      await expect(canvas.getAllByRole("navigation")).toHaveLength(1);
+      await expect(canvas.queryAllByRole("listitem")).toHaveLength(3);
+      await expect(canvas.queryAllByRole("link")).toHaveLength(3);
+    });
+  },
+};
+
+export const NoApps: Story = {
+  args: {
+    appIcons: [],
+    isLoading: false,
+    selectedAppName: "saasure",
+  },
+  render: (props) => {
+    return (
+      <div style={{ height: "100vh" }}>
+        <AppSwitcher {...props} />
+      </div>
+    );
+  },
+  play: async ({ canvasElement, step }: PlaywrightProps<AppSwitcherProps>) => {
+    const canvas = within(canvasElement);
+
+    await step("App switcher shows no items, but is loaded", async () => {
+      await expect(canvas.getAllByRole("navigation")).toHaveLength(1);
+      await expect(canvas.queryAllByRole("listitem")).toHaveLength(0);
+      await expect(canvas.queryAllByRole("link")).toHaveLength(0);
+    });
```

Comment:
You can check on the navigation "to be visible". I wonder if navigations should have a label because someone's not gonna know what they're selecting. I think there's a difference between primary and secondary navigations in ARIA. Something you'll wanna check into.